MELA/interface/TUtil.hh

```diff
@@ -207,7 +208,9 @@ namespace TUtil{
   void GetMassWidth(int ipart, double& m, double& ga);
   void GetMassWidth(const MELAParticle* part, double& m, double& ga);
   void SetCKMElements(double* invckm_ud, double* invckm_us, double* invckm_cd, double* invckm_cs, double* invckm_ts, double* invckm_tb, double* invckm_ub=0, double* invckm_cb=0, double* invckm_td=0);
+  void SetMadgraphCKMElements(double ckmlambda=-1, double ckma=-1, double ckmrho=-1, double ckmeta=-1);
```

Comment:
1) What is the reason to prefer the Wolfenstein parametrization instead of the standard?
2) What is the utility to pick a nonzero eta?
3) Why are the defaults all at -1 and not 0?

Reply:
Regarding (2), what I am really asking is whether complex CKM element are needed at all.

Reply:
1. This is using a function that Madgraph uses to change its ckm matrix values. Direct editing is not possible, as the values would be overridden anyways through the `update_all_params` function.
2. Is there any reason to limit the user? If they never choose to do it they won't. 
   * as an aside, these are the default SM values listed in the parameter card. I could make those the default.
   ```Block ckmblock 
    2 2.265000e-01 # CKMlambda 
    3 7.900000e-01 # CKMA 
    4 1.410000e-01 # CKMrho 
    5 3.570000e-01 # CKMeta 
   ```
3. I can make the defaults the normal SM value from the datacard.

Reply:
I think the defaults here should be motivated by the physics we are doing. Can you derive the default values of these parameters that correspond to the defaults in JHUGen?

Reply:
At the values I provided, these are the default values (sourced from `mod_parameters.F90` and the madgraph CKM function).

| Element | JHU | MG |
| ----------- | ----- | ------ |
| UD | 0.974285d0 | 0.974348875+0i |
| US | 0.225290d0 | 0.2265+0i |
| CS | 0.9734244d0 | 0.974348875+0i |
| CD | -0.225182d0 | -0.2265+0i |
| TB | 0.99912367d0 | 1 + 0i | 
| TS | -0.040920069d0 | -0.0405287775+0j |

The other terms in the matrix (at least for JHUGen) are sourced from these defaults, so I am running under the (probably valid) assumption that the author of SMEFTSIM did the calculations correctly.

Everything seems to agrees to a few decimal points.

Reply:
You can find the values of the CKM elements at mod_parameters.F90 line 2773

Reply:
I know what they are in mod_Parameters :)

A default of ckmlambda=-1, for instance, does not make physical sense. That would (should have) set Vud=Vcs=0.5, Vus=-Vcd = -1 if you use the parametrization in PDG; experimentally measured values are much more different. Leaving aside the fact that it is quite a bad convention to keep -1 as a replacement for defaults in this case, it is also not yet clear whether the PDG parametrization is used.

In the interest of making sure that ME calculations agree as much as possible, I would suggest to find the defaults of these parameters that correspond to what is in JHUGen. Complex CKM values are nice to have, but they create unnecessary differences between MEs that have little physical/experimental importance in Higgs physics.

Reply:
Right - I set the defaults to what gets you the values in the table. The latest PR has this update.

Reply:
Sorry, latest commit I mean.

Reply:
Ok, almost there then. I suggest setting ckmeta=0 by default to keep the MEs as closely as possible. That should ensure Vtd and Vub, which you don't list above, are real. Their values are supposed to be small, <0.01 in absolute value, so their contribution to the full process is negligible.

Reply:
I set cmkmeta to be 0